### DISCUSSION TITLE
Version 0.6.1: How to dynamically synchronize plots without restarting docker-compose

### DISCUSSION THREAD

**Discussion by raingggg** [guydavis/machinaris#372] [Q&A] [answered]
OS: Ubuntu 20.0
Version: 0.6.1

I am not using the machinaris-plotting feature, and instead continue to use my previous plot files/folder sync script. However, the machinaris webui does not update the plots count when new plot files been synced into mounted volumes. Could you please share us how to trigger the plots count updating feature?

Detailed information are listed below:
1. Win10 plotting new files and then a robocopy script will sync them to machinaris controller's disks - say /media/user/plots1 disk in machinaris controller machine
2. /media/user/plots1 has been mounted to the docker volumes like /media/user/plots1:/plots1
3. Expected: webui show the current available plot files, and it will increase step by step since win10 will sync new coming files 
Actual: webui keeps showing the inital available plot files count. I have to run `docker-compose stop && docker-compose up -d` to trigger the updating

Helps needed: please kindly let me know if there is any configuration missing for this case? It would be good to hear if some script can help trigger the update as well. Thank you!

**Comment by guydavis** [comment, marked as answer]
Welcome!  The Machinaris WebUI actually gets the plots listing by asking the Chia farming service, via RPC.  So the WebUI is just reporting what Chia farming service sees as plots.  

Few things to check:

1. Do you have the Machinaris container [configured for each volume mount](https://github.com/guydavis/machinaris/wiki/FAQ#can-i-use-two-or-more-final-destination-directories-to-hold-my-plots) where you are storing plots?  Check the `volumes` line of your docker-compose.yml.  Ensure `plots_dir` in the `environment` sections includes all the in-container paths.
1. From an in-container shell (`docker exec -it machinaris bash`), check that Chia reports correct paths with `chia plots show`
1. There is a refresh frequency check for Chia farming service.  Default is 120 seconds, but please check yours.  Go to Settings | Farming, page choose your Worker by name, and then Fork (probably  Chia).  Search in the config for "frequency" and look to see the current value in seconds.  If not 120, then set that and restart the container to pick up the change:

![image](https://user-images.githubusercontent.com/12936/139087398-428a0a5d-b629-4ac8-ad75-72bab6f295f7.png)

Hope this helps!

**Comment by raingggg** [comment]
@guydavis thanks so much for your quick reply!

Here is my detailed case:
1. Do you have the Machinaris container...  - Yes, it's correct
2. From an in-container shell (docker exec -it machinaris bash)... - Yes, it's correct
3. There is a refresh frequency check for Chia farming service... - Partially Yes, and it should be correct
3.1 In latest machinaris > settings > farming tab, it is actually showing:
` plots_refresh_parameter:
    batch_size: 300
    batch_sleep_milliseconds: 1
    interval_seconds: 120
    retry_invalid_seconds: 1200`
3.2 Above configuration should be correct since chia 1.2.9 uses plots_refresh_parameter instead now:
harvster.py:
![refresh](https://user-images.githubusercontent.com/93122768/139171031-9287c280-e469-4a1c-83a0-2ceac5b725ae.jpg)

BTW, I also tried to list and count both the outer(host) and inner(container) mounted volumes. The count is correct! 

However, I still cannot figure out the root cause. I am planning to check the logs/debug.log and time.time() later to see why.

**Reply by Bafff** [reply]
Hi!

@raingggg I have the same issue but in a completely different situation - docker container with some server software does not see changes in the host linux OS

So I think that something wrong with docker. I still could not find how to fix. Maybe dowgrading Docker would help. Can you check that and reply back?

I dont' think that Chia/Machinaris are the issue

**Comment by Bafff** [comment]
@raingggg any findings?
